### PR TITLE
prevent i18n flickering on pages

### PR DESCRIPTION
**Pull request by KATT**
## What does this PR do?

Prevent i18n flickering on pages

- `/cancel`
- `/success`
- 404. **Note that this will have "flicker of English-content".** See https://github.com/vercel/next.js/discussions/12477 for more details, I don't know of a good solution for this as you can't even provide `getStaticPaths` on 404-pages.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Load 404 page
- [ ] Cancel an event
- [ ] Book an event

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
       "jest-playwright-preset",
       "expect-playwright"
     ],
-    "allowJs": false,
+    "allowJs": true,
```
Comment (KATT) [inline tsconfig.json:41]: Otherwise `import { i18n } from "next-i18next.config";` errors

File: server/lib/ssg.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import { appRouter } from "@server/routers/_app";
+import { createSSGHelpers } from "@trpc/react/ssg";
+
+/**
+ * Initialize static site rendering tRPC helpers.
+ * Provides a method to prefetch tRPC-queries in a `getStaticProps`-function.
+ * Automatically prefetches i18n based on the passed in `context`-object to prevent i18n-flickering.
+ * Make sure to `return { props: { trpcState: ssr.dehydrate() } }` at the end.
+ */
+export async function ssgInit<TParams extends { locale?: string }>(opts: GetStaticPropsContext<TParams>) {
```
Comment (KATT) [inline server/lib/ssg.ts:17]: This function is taken from the pending #1053

File: pages/404.tsx [hunk 3]
```diff
@@ -170,3 +173,13 @@ export default function Custom404() {
     </>
   );
 }
+
+export const getStaticProps = async (context: GetStaticPropsContext) => {
+  const ssr = await ssgInit(context);
```
Comment (KATT) [inline pages/404.tsx:178]: This function will prefetch English locale into the hydrated cache

**Review by KATT — COMMENTED** [review]
Comments to guide reviewers

File: server/lib/ssg.ts (hunk 2 above)
Comment (PeerRich) [reply server/lib/ssg.ts:17]: whats missing for this pending PR to go live?

**Review by PeerRich — APPROVED** [review]
tested visually, works! great stuff

**Pull request merged**